src/parsers/espresso.js

```diff
@@ -25,6 +30,468 @@ function toEspressoFormat(materialOrConfig) {
+    const cellParamsMatch = cellParamsRegex.exec(cards);
+
+    // Extract the units
+    const units = cellParamsMatch[1];
+    let cellUnits = 1.0;
+
+    if (units === "bohr") {
+        if (alat !== null) {
+            throw new Error(
+                "Lattice parameters given in &SYSTEM celldm/A and CELL_PARAMETERS bohr",
+            );
+        }
+        cellUnits = coefficients.BOHR_TO_ANGSTROM;
+    } else if (units === "angstrom") {
+        if (alat !== null) {
+            throw new Error(
+                "Lattice parameters given in &SYSTEM celldm/A and CELL_PARAMETERS angstrom",
+            );
+        }
+        cellUnits = 1.0;
+    } else if (units === "alat") {
+        if (alat === null) {
+            throw new Error("Lattice parameters must be set in &SYSTEM for alat units");
+        } else cellUnits = alat * coefficients.BOHR_TO_ANGSTROM;
+    } else if (alat === null) {
+        cellUnits = coefficients.BOHR_TO_ANGSTROM;
+    } else {
+        cellUnits = alat;
+    }
+
+    const cellParamsData = cellParamsMatch[3].trim().split("\n");
+
+    const cellVectors = [];
+    cellParamsData.forEach((line) => {
+        const values = line
+            .trim()
+            .match(/([\d.-]+)/g)
+            .map(Number)
+            .map((value) => value * cellUnits);
+        cellVectors.push(values);
+    });
+
+    // eslint-disable-next-line no-param-reassign
+    alat = 1; // TODO: remove and change this
+    return {
+        cell: cellVectors,
+        alat,
+        units,
+    };
+}
+
+/**
+ * @summary Creates cell from ibrav and celldm(i) parameters
+ * @param {Object} system
+ * @returns {Number[][], Number, String, String}
+ */
+function ibravToCell(system) {
+    let alat = 1.0;
+    let cell,
+        type,
+        sinab,
+        sinac,
+        tx,
+        ty,
+        tz,
+        a_prime,
+        u,
+        v,
+        v3,
+        b_over_a,
+        c_over_a,
+        cosab,
+        cosac,
+        cosbc;
+    if (system.celldm[1] && system.a) {
+        throw new Error("do not specify both celldm and a,b,c!");
+    } else if (system.celldm[1]) {
+        // celldm(x) in bohr
+        // eslint-disable-next-line prefer-destructuring
+        alat = system.celldm[1] * coefficients.BOHR_TO_ANGSTROM; // this value given in Bohr (a.u.) as per QE specs
+        b_over_a = system.celldm[2] || 0.0;
+        c_over_a = system.celldm[3] || 0.0;
+        cosab = system.celldm[4] || 0.0;
+        cosac = system.celldm[5] || 0.0;
+        cosbc = 0.0;
+        if (system.ibrav === 14) {
+            cosbc = system.celldm[4] || 0.0;
+            cosac = system.celldm[5] || 0.0;
+            cosab = system.celldm[6] || 0.0;
+        }
+    } else if (system.a) {
+        // a, b, c, cosAB, cosAC, cosBC in Angstrom
+        throw new Error("params_to_cell() does not yet support A/B/C/cosAB/cosAC/cosBC");
+    } else {
+        throw new Error("Missing celldm(1)");
+    }
+
+    switch (system.ibrav) {
```

Comment:
I think the real mapping here should be just for the ibrav numbers and lattice types, but the cell definition for lattice types should be inside Lattice, or Cell, perhaps.